### PR TITLE
Feature/include icons

### PR DESCRIPTION
**Pull request by KamsiyonnaObi**
@brandonetter Hey Brandon I am having trouble displaying my project images.

### CODE REVIEW

File: constants/projects.json [hunk 1]
```diff
@@ -1,16 +1,20 @@
 {
     "data": [
         {
-            "img": "https://example.com",
             "title": "Morrent",
+            "mobileImg": "@/public/project-images/Morrent/Morrent-Mobile.png",
+            "laptopImg": "/public/project-images/Morrent/Morrent-Laptop.png",
```
Comment (brandonetter) [inline constants/projects.json]:
```suggestion
            "laptopImg": "/project-images/Morrent/Morrent-Laptop.png",
```

"public" is the folder that gets exposed, so we shouldn't include it :)

File: constants/projects.json [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
             "desc": "A Car Rental Application",
             "frontEnd": ["ReactJS","NextJS"],
             "backEnd": ["NodeJS", "MongoDB"],
             "color": "#416CEA"
         },
         {
-            "img": "https://example.com",
             "title": "JobIT",
+            "mobileImg": "@/public/project-images/Jobit/JobIT-Mobile.png",
+            "laptopImg": "/public/project-images/Jobit/JobIT-Laptop.png",
```
Comment (brandonetter) [inline constants/projects.json]:
```suggestion
            "laptopImg": "/project-images/Jobit/JobIT-Laptop.png",
```

File: constants/projects.json [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
+            "imgAlt": "Morrent",
             "desc": "A Car Rental Application",
             "frontEnd": ["ReactJS","NextJS"],
             "backEnd": ["NodeJS", "MongoDB"],
             "color": "#416CEA"
         },
         {
-            "img": "https://example.com",
             "title": "JobIT",
+            "mobileImg": "@/public/project-images/Jobit/JobIT-Mobile.png",
```
Comment (brandonetter) [inline constants/projects.json]:
```suggestion
            "mobileImg": "/project-images/Jobit/JobIT-Mobile.png",
```

File: constants/projects.json [hunk 4]
```diff
@@ -1,16 +1,20 @@
 {
     "data": [
         {
-            "img": "https://example.com",
             "title": "Morrent",
+            "mobileImg": "@/public/project-images/Morrent/Morrent-Mobile.png",
```
Comment (brandonetter) [inline constants/projects.json]:
```suggestion
            "mobileImg": "/project-images/Morrent/Morrent-Mobile.png",
```

**Review by brandonetter — CHANGES_REQUESTED** [review]
This naming convention is a little confusing. This is a 'card':
image

But you have a component called "ProjectCard" which is actually the section that maps over multiple cards. And you already have something for the project section called `Projects.tsx`

I would rework your ProjectCard to be an individual card, and just map over them inside of `Projects`.

**Comment by brandonetter** [comment]
Also reworking your image display could help you out. You're using a div around the image to control the styling- but you're not actually making the image fit the div.

Use fill instead and object-contain, you'll end up with an image that becomes the background of that div. This will let you control it better (it already fits and centers with just this change)
```js
    <div className="relative h-[155px] border lg:h-[331px]">
                        <Image src={project.laptopImg} className="object-contain" fill alt={project.imgAlt}/>
                    </div>
```

image


## Overflow

If you want to do what the design does- and cut-off part of the image. Add these two changes 
1) Add a margin to the image
2) Add `overflow-hidden` to the div.

```js
   <div className="relative h-[155px] border lg:h-[331px] overflow-hidden">
                        <Image src={project.laptopImg} className="ml-24 object-contain" fill alt={project.imgAlt}/>
                    </div>
```

image

Obviously this works better with the mobile image included :)

**Comment by KamsiyonnaObi** [comment]
> This naming convention is a little confusing. This is a 'card': image
> 
> But you have a component called "ProjectCard" which is actually the section that maps over multiple cards. And you already have something for the project section called `Projects.tsx`
> 
> I would rework your ProjectCard to be an individual card, and just map over them inside of `Projects`.

Hey, thanks for the feedback. I don't quite understand what you mean here.

Do you mean map over data(projects) and then pass each project as a prop down to the project Card?

**Comment by brandonetter** [comment]
@KamsiyonnaObi 

I mean to have your `ProjectCard.tsx` file actually be a "Project Card", and for your `Projects.tsx` file to actually be your "Projects". 

It doesn't make much sense to have your `page.tsx` include a `projects` component that is really just the wrapper for the section, and then it includes a `ProjectCard` component that is the actual section containing **multiple** cards.

This is just about clarity in naming, and also about not nesting all of your logic multiple components deep when it isn't necessary.

So I suggest your `ProjectCard.tsx` just be each card:
```js
const ProjectCard = ({project,pcolor,swap}:Props) =>(
<div style={pcolor} className={`flex flex-col ${swap ? 'lg:flex-row-reverse' : ''} w-[345px] h-fit mx-auto justify-center rounded-[20px] gap-6 py-[46px] lg:w-fit lg:flex-row lg:gap-[30.89px]`}>
    <div className="flex flex-col self-center content-center gap-[30px] px-6 lg:gap-6">
        <div>
            <h1 className="text-white-900 text-[32px] leading-[36.8px] tracking-[-0.36px] font-bold lg:header2">
                {`${project.title} - `}<br/>{` ${project.desc} `}
            </h1>
        </div>
        <div className="flex gap-5">
            <div className="p-2.5 bg-white-900 bg-opacity-20 rounded-md">
                <p className="inline lg-reg">{project.frontEnd?.join(', ')}</p>
            </div>
            <div className="p-2.5 bg-white-900 bg-opacity-20 rounded-md">
                <p className="inline lg-reg">{project.backEnd?.join(', ')}</p>
            </div>
        </div>
        <div>
            <Link href={"/"}>
                <p className="body-bold mt-[26px]">See Project Details</p>
            </Link>
        </div>
    </div>
    <div className="w-[320px] pl-6 lg:w-[585px]">
        <div className="relative h-[155px] border lg:h-[331px] overflow-hidden">
            <Image src={project.laptopImg} className="ml-24 object-contain" fill alt={project.imgAlt}/>
        </div>
    </div>
</div>
)
```

And I suggest your `Projects.tsx` actually do the mapping to create the "Projects":
```js
import projects from '../constants/projects.json';
import Link from 'next/link';

const Projects = () => {
  return (
    <article className="flex flex-col justify-center gap-9 lg:gap-12">
        {/* Heading */}
        <div className="flex mx-auto">
            <h1 className="text-black-200 text-[36px] font-bold sm:header2 dark:text-white-900">
                Featured <span className="highlight">Projects</span>
            </h1>
        </div>
        {/* Project Cards */}
        <div className="flex gap-9 flex-wrap sm:justify-center lg:gap-12">
        {projects.data.map((project, idx) => {
            const swap = idx % 2;
            const pcolor = {
                backgroundColor: project.color,
            };
            return <ProjectCard
            project={project}
            key={project.title}
            swap={swap}
            pcolor={pcolor}
            />
        })}
        </div>
        {/* See More */}
        <Link
            href={"/"}
            className="hidden body-bold py-5 px-[45.5px] rounded-full self-center bg-Accent-light dark:bg-Accent-dark sm:flex"
        >
            See more case studies
        </Link>
   </article>
  )
}
```

**Comment by brandonetter** [comment]
You can keep it the way you have it- but then I would suggest you rename them. It's typical to create a single component for a card like this and map over it, it's less typical to map over large 25+ line section of JSX

**Comment by brandonetter** [comment]
image

This line break `br` isn't necessary to match the design. Was this intentional? It seems like you're doing this just to match the mobile design, but it now doesn't match the desktop design.

Instead, try using a breakpoint to change if the text is a "block" style element (it takes up it's own line), or an "inline" style element (it continues on the same line as any previous element).

```js
<h1 className=" text-white-900 text-[32px] leading-[36.8px] tracking-[-0.36px] font-bold lg:header2">
                {`${project.title} - `}
                <span className="inline-block lg:inline">
                    {` ${project.desc} `}
                </span>
            </h1>
```

**Pull request merged**